src/main/menu.main.ts

```diff
@@ -1,525 +1,54 @@
-
-            // About menu
-            const aboutMenuAdditions: MenuItemConstructorOptions[] = [
-                { type: 'separator' },
-            ];
-
-            if (!isWindowsStore() && !isSnapStore()) {
-                aboutMenuAdditions.push(updateMenuItem);
-            }
-
-            aboutMenuAdditions.push({
-                label: this.i18nService.t('aboutBitwarden'),
-                click: async () => {
-                    const aboutInformation = this.i18nService.t('version', app.getVersion()) +
-                        '\nShell ' + process.versions.electron +
-                        '\nRenderer ' + process.versions.chrome +
-                        '\nNode ' + process.versions.node +
-                        '\nArchitecture ' + process.arch;
-                    const result = await dialog.showMessageBox(this.windowMain.win, {
-                        title: 'Bitwarden',
-                        message: 'Bitwarden',
-                        detail: aboutInformation,
-                        type: 'info',
-                        noLink: true,
-                        buttons: [this.i18nService.t('ok'), this.i18nService.t('copy')],
-                    });
-                    if (result.response === 1) {
-                        clipboard.writeText(aboutInformation);
-                    }
-                },
-            });
-
-            template[template.length - 1].submenu =
-                (template[template.length - 1].submenu as MenuItemConstructorOptions[]).concat(aboutMenuAdditions);
-        }
-
-        (template[template.length - 2].submenu as MenuItemConstructorOptions[]).splice(1, 0,
-            {
-                label: this.main.i18nService.t(process.platform === 'darwin' ? 'hideToMenuBar' : 'hideToTray'),
-                click: () => this.main.messagingService.send('hideToTray'),
-                accelerator: 'CmdOrCtrl+Shift+M',
-            },
-            {
-                type: 'checkbox',
-                label: this.main.i18nService.t('alwaysOnTop'),
-                checked: this.windowMain.win.isAlwaysOnTop(),
-                click: () => this.main.windowMain.toggleAlwaysOnTop(),
-                accelerator: 'CmdOrCtrl+Shift+T',
-            });
-        this.menu = Menu.buildFromTemplate(template);
-        Menu.setApplicationMenu(this.menu);
+    private async setMenu(updateRequest?: MenuUpdateRequest) {
+        Menu.setApplicationMenu(new Menubar(
+            this.main.i18nService,
+            this.main.messagingService,
+            this.main.updaterMain,
+            this.windowMain,
+            await this.getWebVaultUrl(),
+            app.getVersion(),
+            updateRequest,
+        ).menu);
     }
 
-    private async openWebVault() {
-        let webUrl = 'https://vault.bitwarden.com';
-        const urlsObj: any = await this.main.storageService.get(ConstantsService.environmentUrlsKey);
+    private async getWebVaultUrl() {
+        let webVaultUrl = cloudWebVaultUrl;
```

Comment:
Do we still need this fallback after the env urls refactor?

I believe we can replace ths logic with `getWebVaultUrl`, which we can use in whatever menu uses it vs here.

Reply:
the env urls refactor only currently has a default for server. I think we should leave this as is for now, that's a pretty fragile area that doesn't seem wise to break today.

Reply:
@addisonbeck could you add a tech debt item to this please? We should only have this kind of logic in env service.

Reply:
https://app.asana.com/0/1201461753012676/1201521717039120